src/memory/virtualMemory.ts

```diff
@@ -0,0 +1,244 @@
+import { PageTable } from './pageTable';
+import { AddressIndex } from './addressIndex';
+import { Segment } from './segment';
+import { SegmentAddress } from './segmentAddress';
+import { MemoryError, MemoryErrorType } from './memoryError';
+
+export class VirtualMemory {
+  readonly l1PageTable: PageTable = new PageTable();
```

Comment:
These should be private but I just made it public for now because it's easier to test. Anyway it's readonly.